lib/improver/tests/nbhood/test_squareneighbourhood.py

```diff
@@ -232,16 +232,71 @@ def test_basic(self):
         self.assertIsInstance(cube, Cube)
         self.assertArrayAlmostEqual(result.data, data)
 
-    def test_masked_array_fail(self):
-        """Test that the correct exception is raised when a masked array
-           is passed in."""
+    def test_masked_array(self):
+        """Test that the run method produces a cube with correct data when a
+           cube containing masked data is passed in."""
         cube = set_up_cube(
             zero_point_indices=((0, 0, 2, 2),), num_time_points=1,
             num_grid_points=5)
-        cube.data = np.ma.masked_equal(cube.data, 1)
-        msg = 'Masked data is not currently supported'
-        with self.assertRaisesRegexp(ValueError, msg):
-            SquareNeighbourhood().run(cube, self.RADIUS)
+        cube.data = np.array([[[[1, 1, 0, 1, 1],
```

Comment:
Yes, you are correct, the summation is setting everything in the sum after the NaN to NaN. 

This could be resolved by numpy's `nancumsum` function (see https://docs.scipy.org/doc/numpy-dev/reference/generated/numpy.nancumsum.html) but this isn't available in the version of numpy we are using, unfortunately.

In the meantime, we could set the NaNs to 0 before doing the summation, then reapply a mask of NaNs to the output data?